plugins/ReverbSC/ReverbSC.cpp

```diff
@@ -0,0 +1,141 @@
+/*
+ * ReverbSC.cpp - A native reverb based on an algorithm by Sean Costello
+ *
+ * This file is part of LMMS - http://lmms.io
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program (see COPYING); if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA.
+ *
+ */
+
+#include <math.h>
+#include "ReverbSC.h"
+
+#include "embed.cpp"
+
+#define DB2LIN(X) pow(10, X / 20.0f);
+
+extern "C"
+{
+
+Plugin::Descriptor PLUGIN_EXPORT reverbsc_plugin_descriptor =
+{
+	STRINGIFY( PLUGIN_NAME ),
+	"ReverbSC",
+	QT_TRANSLATE_NOOP( "pluginBrowser", "Reverb algorithm by Sean Costello" ),
+	"Paul Batchelor",
+	0x0123,
+	Plugin::Effect,
+	new PluginPixmapLoader( "logo" ),
+	NULL,
+	NULL
+} ;
+
+}
+
+ReverbSCEffect::ReverbSCEffect( Model* parent, const Descriptor::SubPluginFeatures::Key* key ) :
+	Effect( &reverbsc_plugin_descriptor, parent, key ),
+	m_reverbSCControls( this )
+{
+	sp_create(&sp);
+	sp->sr = Engine::mixer()->processingSampleRate();
+
+	sp_revsc_create(&revsc);
+	sp_revsc_init(sp, revsc);
+
+	sp_dcblock_create(&dcblk[0]);
+	sp_dcblock_create(&dcblk[1]);
+	
+	sp_dcblock_init(sp, dcblk[0]);
+	sp_dcblock_init(sp, dcblk[1]);
+}
+
+ReverbSCEffect::~ReverbSCEffect()
+{
+	sp_revsc_destroy(&revsc);
+	sp_dcblock_destroy(&dcblk[0]);
+	sp_dcblock_destroy(&dcblk[1]);
+	sp_destroy(&sp);
+}
+
+bool ReverbSCEffect::processAudioBuffer( sampleFrame* buf, const fpp_t frames )
+{
+	if( !isEnabled() || !isRunning () )
+	{
+		return( false );
+	}
+
+	double outSum = 0.0;
+	const float d = (dryLevel() + 1) * 0.5;
+	const float w = (wetLevel() + 1) * 0.5;
+
+	SPFLOAT tmpL, tmpR;
+	SPFLOAT dcblkL, dcblkR;
+	
+	ValueBuffer * inGainBuf = m_reverbSCControls.m_inputGainModel.valueBuffer();
+	ValueBuffer * sizeBuf = m_reverbSCControls.m_sizeModel.valueBuffer();
+	ValueBuffer * colorBuf = m_reverbSCControls.m_colorModel.valueBuffer();
+	ValueBuffer * outGainBuf = m_reverbSCControls.m_outputGainModel.valueBuffer();
+
+	for( fpp_t f = 0; f < frames; ++f )
+	{
+		outSum += buf[f][0]*buf[f][0] + buf[f][1]*buf[f][1];
```

Comment:
Shouldn't outsum be an absolute number like:
`outSum += fabs( buf[f][0]*buf[f][0] ) + fabs( buf[f][1]*buf[f][1] );`

If I interpret this correctly the [checkGate function](https://github.com/LMMS/lmms/blob/48cc3bb5ef85bac8cc7c01b11dd3139c547a0677/src/core/Effect.cpp#L140:L162) basically just tests if the signal is lower than the gate value over one frame, so if the input is a wave with a period that is divisible with the buffer size it would see an input of 0, silence and cut out.